packages/catalog-process/src/services/catalogService.ts

```diff
@@ -165,113 +179,60 @@ const deprecateDescriptor = async (
   );
 };
 
-const hasNotDraftDescriptor = (eService: EService): boolean => {
-  const hasNotDraftDescriptor = eService.descriptors.some(
-    (d: Descriptor) => d.state === descriptorState.draft,
-    0
+const hasNotDraftDescriptor = (eService: EService): void => {
+  const hasDraftDescriptor = eService.descriptors.some(
+    (d: Descriptor) => d.state === descriptorState.draft
   );
-  if (!hasNotDraftDescriptor) {
+  if (hasDraftDescriptor) {
     throw draftDescriptorAlreadyExists(eService.id);
   }
-  return hasNotDraftDescriptor;
 };
```

Comment:
I changed this logic because we should throw an error if a draft descriptor exist